src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1217,7 +1217,10 @@ public static VariantContext simpleMerge(final Collection<VariantContext> unsort
 
         final String ID = rsIDs.isEmpty() ? VCFConstants.EMPTY_ID_FIELD : Utils.join(",", rsIDs);
 
-        final VariantContextBuilder builder = new VariantContextBuilder().source(name).id(ID);
+        // This preserves the GATK3-like behavior of reporting all sources, delimited with hyphen:
+        final String allSources = variantSources.isEmpty() ? name : variantSources.stream().sorted().collect(Collectors.joining("-"));
```

Comment:
It's a good question.  Apparently this is what GATK3 did.  @bbimber Any objection to capping it at some reasonable length?

Reply:
Honestly, I dont have personal use-cases around this, but on DISCVR-seq (which ported GATK3's CombineVariants) I get a surprising amount of requests from people around the lack-of or changes to this feature. 

I have no real objection to some kind of cap on length. What length would you find reasonable? I suppose that value could be parameterized, but this would require exposing an argument to simpleMerge() and maybe overloading that method.

Anyway, this feels like a valid but fringe case to me. I dont have strong opinions one way or another.

Reply:
Combining hundreds or thousands of VCF records is something we actually do frequently in practice. I think a parameterized option would be overkill, but some kind of sensible hardcoded max length for the source field seems appropriate.

Reply:
@lbergelson I defer to you to do something reasonable in this PR to address the effects of this change on VCF file size for large cohorts.

Reply:
FWIW, after thinking O/N on this I would probably argue for overloading the method with one that adds arguments for "storeAllVcfSources", and "maxSourceLength", but keep the existing signature where storeAllVcfSources=False, and maxSourceLength=-1 (which is moot since the existing behavior of taking the first source would remian). this preserves the status quo, but callers could opt-in to GATK3-like behavior. I am happy to make those changes.